Error while checking build/O2/fullCI_slc9 for 56894e1b7a5a33c189d1f543631da71bad5b7187 at 2025-05-13 23:59:
```
## sw/BUILD/arrow-latest/log
/sw/BUILD/5c423fbe2eb3b52187e935043b0d4602548f9a00/arrow/src_tmp/cpp/src/gandiva/precompiled/arithmetic_ops.cc:18:10: fatal error: 'cmath' file not found
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14237/56894e1b7a5a33c189d1f543631da71bad5b7187/build_O2_fullCI_slc9/pretty.html).

--- a/Detectors/EMCAL/base/include/EMCALBase/Geometry.h
+++ b/Detectors/EMCAL/base/include/EMCALBase/Geometry.h
@@ -22,7 +22,9 @@
 #include <TNamed.h>
 #include <TParticle.h>
 #include <TVector3.h>
+#include <TObjArray.h>
 
+#include "CCDB/BasicCCDBManager.h"
 #include "DataFormatsEMCAL/Constants.h"
 #include "EMCALBase/GeometryBase.h"
 #include "MathUtils/Cartesian.h"
@@ -57,7 +59,7 @@ class Geometry
   /// | EMCAL_COMPLETE12SMV1_DCAL             | Full EMCAL, 10 DCAL Supermodules (not used in practice)  |
   /// | EMCAL_COMPLETE12SMV1_DCAL_8SM         | Full EMCAL, 8 DCAL Supermodules (run2)                   |
   /// | EMCAL_COMPLETE12SMV1_DCAL_DEV         | Full EMCAL, DCAL development geometry (not used)         |
-  Geometry(const std::string_view name, const std::string_view mcname = "", const std::string_view mctitle = "");
+  explicit Geometry(const std::string_view name, const std::string_view mcname = "", const std::string_view mctitle = "");
 
   /// \brief Copy constructor.
   Geometry(const Geometry& geom);
@@ -563,6 +565,11 @@ class Geometry
   /// Move from header due to coding violations : Dec 2,2011 by PAI
   ///
   void SetMisalMatrix(const TGeoHMatrix* m, Int_t smod) const;
+
+  ///
+  /// Method to set shift-rotational matrixes from CCDB
+  ///
+  void SetMisalMatrixFromCcdb(const char* path = "Users/m/mhemmer/EMCAL/Config/GeometryAligned", int timestamp = 10000) const;
 
   ///
   /// Transform clusters cell position into global with alternative method, taking into account the depth calculation.

--- a/Detectors/EMCAL/base/src/Geometry.cxx
+++ b/Detectors/EMCAL/base/src/Geometry.cxx
@@ -8,16 +8,21 @@
 // In applying this license CERN does not waive the privileges and immunities
 // granted to it by virtue of its status as an Intergovernmental Organization
 // or submit itself to any jurisdiction.
+#include "EMCALBase/Geometry.h"
+
+#include <fairlogger/Logger.h>
+
 #include <iomanip>
+#include <string>
+#include <algorithm>
+#include <cstdio>
+#include <tuple>
 
 #include <TGeoBBox.h>
 #include <TGeoManager.h>
 #include <TGeoMatrix.h>
 #include <TList.h>
 
-#include <fairlogger/Logger.h>
-
-#include "EMCALBase/Geometry.h"
 #include "EMCALBase/ShishKebabTrd1Module.h"
 
 #include <boost/algorithm/string/predicate.hpp>
@@ -1557,6 +1562,7 @@ const TGeoHMatrix* Geometry::GetMatrixForSuperModule(Int_t smod) const
 
   if (!SMODULEMATRIX[smod]) {
     if (gGeoManager) {
+      LOG(info) << "Loading EMCAL misalignment matrix for SM " << smod << " from GeoManager.";
       SetMisalMatrix(GetMatrixForSuperModuleFromGeoManager(smod), smod);
     } else {
       LOG(fatal) << "Cannot find EMCAL misalignment matrices! Recover them either: \n"
@@ -1759,6 +1765,25 @@ void Geometry::SetMisalMatrix(const TGeoHMatrix* m, Int_t smod) const
     }
   } else {
     LOG(fatal) << "Wrong supermodule index -> " << smod << std::endl;
+  }
+}
+
+void Geometry::SetMisalMatrixFromCcdb(const char* path, int timestamp) const
+{
+  LOG(info) << "Using CCDB to obtain EMCal alignment.";
+  o2::ccdb::CcdbApi api;
+  map<string, string> metadata; // can be empty
+  api.init("http://alice-ccdb.cern.ch");
+  TObjArray* matrices = api.retrieveFromTFileAny<TObjArray>(path, metadata, timestamp);
+
+  for (int iSM = 0; iSM < mNumberOfSuperModules; ++iSM) {
+    TGeoHMatrix* mat = reinterpret_cast<TGeoHMatrix*>(matrices->At(iSM));
+    if (mat) {
+
+      SetMisalMatrix(mat, iSM);
+    } else {
+      LOG(info) << "Could not obtain Alignment Matrix for SM " << iSM;
+    }
   }
 }
 

